src/QMCWaveFunctions/Fermion/MultiSlaterDeterminantFast.cpp

```diff
@@ -1046,20 +1056,30 @@ void MultiSlaterDeterminantFast::evaluateDerivativesWF(ParticleSet& P,
     }
   }
 
-  if (Dets.size() != 2)
-  {
-    throw std::runtime_error(
-        "MultiSlaterDeterminantFast::evaluateDerivativesWF only compatible with two quantum particle types.");
-  }
-  else
+  //Currently, the MultiDiracDeterminant::evaluateDerivativesWF works with a legacy design, essentially requiring only up and down determinants.
+  //e.g. for spinor cases, we only have one determinant so this interface doesn't work.
+  //Here we throw an error only if the optimization is turned on for MultiDiracDeterminants until the code is updated
+  bool all_Optimizable = true;
+  for (size_t id = 0; id < Dets.size() && all_Optimizable; id++)
```

Comment:
as above

Reply:
added new private function evaluateMultiDiracDeterminantDerivativesWF to handle this